docs/tfengine/schemas/cicd.md

```diff
@@ -27,8 +27,8 @@
 | envs.triggers.validate.run_on_push | Whether or not to be automatically triggered from a PR/push to branch.                    Default to true. | boolean | false | - | - |
```

Comment:
This is a tricky one. I haven't updated the `envs` templating because of how it's rendered.

At `triggers.tf`, the whole configuration is wrapped around a giant `range` block, and renders the resource block for every environment. I think I have two ways to proceed (and writing the variable `envs` at `variables.tf`)

1. Continue rendering the giant `range` block but in each resource using `element(list, index)` to access the corresponding env index.
2. Use a `for_each` in each resource.

The problems are:

1. It creates a bunch of unnecessary code and it requires the user to input a list of the same length of arguments used across the `triggers.tf`. Not my favorite but it keeps the envs divided (which is what I think you tried)
2. Seems better for me but I don't know if it's completely possible and also, there will be no separation of envs at the file. However, it's more flexible.

What do you think?

Reply:
We decided to solve it by creating sub modules